hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -590,7 +590,7 @@ private synchronized boolean updateAddress() throws IOException {
       InetSocketAddress currentAddr = NetUtils.createSocketAddrForHost(
                                server.getHostName(), server.getPort());
 
-      if (!server.equals(currentAddr)) {
+      if (!currentAddr.isUnresolved() && !server.equals(currentAddr)) {
         LOG.warn("Address change detected. Old: " + server.toString() +
                                  " New: " + currentAddr.toString());
```

Comment:
Since you are touching this part can you fix this log line as well?
```
        LOG.warn("Address change detected. Old: {} New: {}", server, currentAddr);
```